### ISSUE TITLE
Update requirements: librosa==0.7.2 does not work with joblib==0.12.2

### ISSUE THREAD

**Issue by thorwhalen** [librosa/librosa#1090]
Importing `librosa` (on `iOS 10.14.6`, with `python 3.82`) leads to a `TypeError` (see traceback below). 
Traced the problem to an import of `joblib`. Updating to `` worked for me. Perhaps `librosa` requirements should be updated?

More specifically, the following combination works:
```
librosa==0.7.2
joblib==0.14.1
```

But the combination below does **not work**
```
librosa==0.7.2
joblib==0.12.2
```

It produces the following traceback:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-989066361697> in <module>
----> 1 import librosa

~/.pyenv/versions/3.8.2/lib/python3.8/site-packages/librosa/__init__.py in <module>
      9 
     10 # And all the librosa sub-modules
---> 11 from ._cache import cache
     12 from . import core
     13 from . import beat

~/.pyenv/versions/3.8.2/lib/python3.8/site-packages/librosa/_cache.py in <module>
      4 
      5 import os
----> 6 from joblib import Memory
      7 
      8 

~/.pyenv/versions/3.8.2/lib/python3.8/site-packages/joblib/__init__.py in <module>
    117 from .numpy_pickle import load
    118 from .compressor import register_compressor
--> 119 from .parallel import Parallel
    120 from .parallel import delayed
    121 from .parallel import cpu_count

~/.pyenv/versions/3.8.2/lib/python3.8/site-packages/joblib/parallel.py in <module>
     30                                  LokyBackend)
     31 from ._compat import _basestring
---> 32 from .externals.cloudpickle import dumps, loads
     33 from .externals import loky
     34 

~/.pyenv/versions/3.8.2/lib/python3.8/site-packages/joblib/externals/cloudpickle/__init__.py in <module>
      1 from __future__ import absolute_import
      2 
----> 3 from .cloudpickle import *
      4 
      5 __version__ = '0.5.2'

~/.pyenv/versions/3.8.2/lib/python3.8/site-packages/joblib/externals/cloudpickle/cloudpickle.py in <module>
    149 
    150 
--> 151 _cell_set_template_code = _make_cell_set_template_code()
    152 
    153 

~/.pyenv/versions/3.8.2/lib/python3.8/site-packages/joblib/externals/cloudpickle/cloudpickle.py in _make_cell_set_template_code()
    130         )
    131     else:
--> 132         return types.CodeType(
    133             co.co_argcount,
    134             co.co_kwonlyargcount,

TypeError: an integer is required (got type bytes)
```

**Comment by bmcfee** [comment]
Thanks for catching this.  This appears to be a problem specific to python 3.8 arising in joblib (fixed in 0.14, as noted [here](https://github.com/joblib/joblib/blob/master/CHANGES.rst#release-0140)).

I think your proposed solution is the correct one, but it will need to be rolled into a bigger set of updates to officially add support and testing for python 3.8.

**Comment by thorwhalen** [comment]
Okay. Probably not needed, but [pull request 1091 here](https://github.com/librosa/librosa/pull/1091) anyway.
